File: specs/Qowaiv.CodeAnalysis.CSharp.Specs/DecorateFunctions_specs.cs
namespace Rules.DecorateFunctions_specs;

public class Verify
{
    [Test]
    public void CSharp()
        => new DecorateFunctions()
        .ForCS()
        .AddSource(@"Cases\DecorateFunctions.cs")
        .AddReference<FluentAssertions.CustomAssertionAttribute>()
        .Verify();
}

File: specs/Qowaiv.CodeAnalysis.VisualBasic.Specs/DecorateFunctions_specs.vb
Imports NUnit.Framework
Imports CodeAnalysis.TestTools
Imports Qowaiv.CodeAnalysis

Namespace DecorateFunctions_specs

    Public Class Verify

        <Test>
        Public Sub VisualBasic()
            Dim rule As New DecorateFunctions()
            rule.ForVB().
                AddSource("Cases/DecorateFunctions.vb").
                AddReference(Of Qowaiv.Percentage).
                Verify()
        End Sub

    End Class

End Namespace
